### PR TITLE
enh: switch for event streams + share clean-up

### PR DESCRIPTION
**Pull request by spolu**
@lasryaric this one is for you

@Duncid new share UX

Screenshot from 2023-09-19 21-31-04

### CODE REVIEW

File: front/lib/api/assistant/agent.ts [hunk 1]
```diff
@@ -172,34 +173,41 @@ export async function* runAgent(
...
+            // the agentMessage object, updating this object will update the conversation as well.
+            agentMessage.action = event.action;
+            break;
 
-          // We stitch the action into the agent message. The conversation is expected to include
-          // the agentMessage object, updating this object will update the conversation as well.
-          agentMessage.action = event.action;
+          default:
+            ((blockParent: never) => {
+              logger.error("Unknown `runAgent` event type", blockParent);
```
Comment (lasryaric) [inline front/lib/api/assistant/agent.ts:208]: Can you rename block parent to event or something appropriate?

File: front/lib/api/assistant/agent.ts [hunk 2]
```diff
@@ -220,32 +228,41 @@ export async function* runAgent(
...
+            configurationId: configuration.sId,
+            messageId: agentMessage.sId,
+            text: event.text,
+          };
+
+          agentMessage.content = event.text;
+          break;
+
+        default:
+          ((blockParent: never) => {
```
Comment (lasryaric) [inline front/lib/api/assistant/agent.ts:262]: Can you rename block parent to event or something appropriate?

File: front/lib/api/assistant/conversation.ts [hunk 3]
```diff
@@ -1058,59 +1058,70 @@ async function* streamRunAgentEvents(
...
+
+      // All other events that won't impact the database and are related to actions or tokens
+      // generation.
+      case "retrieval_params":
+      case "generation_tokens":
+        yield event;
+        break;
+
+      default:
+        ((blockParent: never) => {
```
Comment (lasryaric) [inline front/lib/api/assistant/conversation.ts:1118]: Can you rename block parent to event or something appropriate?

**Review by lasryaric — CHANGES_REQUESTED** [review]
Just a few renaming.
I am loving these switch!

**Comment by spolu** [comment]
@lasryaric done! I cargo-culted yours so I fixed yours as well :)

**Review by lasryaric — APPROVED** [review]
😝

**Pull request merged**